crates/api/src/func.rs

```diff
@@ -474,6 +474,46 @@ impl Func {
         func.into_func(store)
     }
 
+    /// Creates a new `Func` for a function which performs a program exit,
+    /// unwinding the stack up the point where wasm was most recently entered.
+    pub fn exit_func(store: &Store) -> Func {
```

Comment:
I'm curious if we can find a bit of a cleaner API to export this?

It looks like host functions have been updated to return `Result<T>` rather than `Result<T, Trap>`, which means that any arbitrary error type can be returned. Could we perhaps special case when the error is `Exit` and propagate that through, removing the need for an extra constructor here?